src/app.ts

```diff
@@ -1,162 +1,181 @@
-      alert(error.message);
+      render.message('error', 'top', error.message);
       return;
     }
 
-    this.$modal.close();
-    this.updateRestaurants();
+    render.message('success', 'bottom', ALERT_MESSAGE.createRestaurant);
+    render.closeRegisterRestaurantModal();
+    this.updateRestaurantsList();
   };
 
-  createRestaurant = (event: Event) => {
-    const restaurantProps = Object.fromEntries([
-      ...new FormData(event.target as HTMLFormElement).entries(),
-    ]);
-
+  createRestaurant = ({
+    category,
+    name,
+    distanceByMinutes,
+    description,
+    referenceUrl,
+  }: RestaurantProps) => {
     return new Restaurant({
-      category: String(restaurantProps.category),
-      name: String(restaurantProps.name),
-      distanceByMinutes: Number(restaurantProps.distanceByMinutes),
-      description: String(restaurantProps.description),
-      referenceUrl: String(restaurantProps.referenceUrl),
+      category,
+      name,
+      distanceByMinutes,
+      description,
+      referenceUrl,
+      existRestaurantsName: this.#restaurants.map((restaurant) => restaurant.getName()),
     });
   };
 
-  initSelect() {
-    this.$restaurantFilterSelect.setOptions(DEFAULT_FILTER_OPTIONS);
-    this.$restaurantSortSelect.setOptions(DEFAULT_SORT_OPTIONS);
-  }
+  openRestaurantDetailModal = ({ detail }: CustomEvent) => {
+    const { name } = detail;
 
-  initModalSelect() {
-    this.$restaurantModalCategory.setOptions(DEFAULT_MODAL_CATEGORY_OPTIONS);
-    this.$restaurantModalDistance.setOptions(DEFAULT_MODAL_DISTANCE_OPTIONS);
-  }
+    const restaurant = this.#restaurants.filter((_restaurant) => _restaurant.getName() === name)[0];
+
+    render.openRestaurantDetailModal(restaurant);
+  };
+
+  initLoad = () => {
+    render.init();
+
+    const restaurants: Restaurant[] = getRestaurants();
+
+    if (!!restaurants.length) {
+      this.#restaurants = restaurants.map((restaurant: Restaurant) =>
+        Object.setPrototypeOf(restaurant, Restaurant.prototype),
+      );
+    }
+
+    this.updateRestaurantsList();
+  };
+
+  deleteRestaurant = ({ detail }: CustomEvent) => {
+    this.#restaurants = this.#restaurants.filter(
+      (restaurant) => restaurant.getName() !== detail.name,
+    );
+
+    render.message('success', 'bottom', ALERT_MESSAGE.removeRestaurant);
+    this.updateRestaurantsList();
+    render.closeRestaurantDetailModal();
+  };
+
+  toggleRestaurantFavorite = ({ detail }: CustomEvent) => {
+    const { restaurantName } = detail;
+
+    this.#restaurants.forEach((restaurant) => {
+      if (restaurant.getName() === restaurantName) restaurant.toggleFavorite();
+    });
+
+    const targetRestaurant = this.#restaurants.find(
+      (restaurant) => restaurant.getName() === restaurantName,
+    );
+
+    if (!targetRestaurant) return errorHandler.doseNotExistRestaurant();
+
+    render.toggleRestaurantFavorite(this.#restaurantListType, targetRestaurant);
+
+    saveRestaurants(this.#restaurants);
+  };
```

Comment:
`forEach`구문과 `targetRestaurant`를 찾는 구문에서 로직에 중복이 보이는데요.

순서를 바꿔 아래처럼 작성해보시는 것도 좋을거 같네요!

```suggestion
toggleRestaurantFavorite = ({ detail }: CustomEvent) => {
  const { restaurantName } = detail;

  const targetRestaurant = this.#restaurants.find(
    (restaurant) => restaurant.getName() === restaurantName,
  );

  if (!targetRestaurant) return errorHandler.doseNotExistRestaurant();

  targetRestaurant.toggleFavorite();

  render.toggleRestaurantFavorite(this.#restaurantListType, targetRestaurant);

  saveRestaurants(this.#restaurants);
};
```

Reply:
`Array.find()` 메서드를 앞으로 많이 사용해봐야겠습니다ㅎㅎ 👍👍👍